src/contracts/core/PaymentCoordinator.sol

```diff
@@ -0,0 +1,286 @@
+            RangePayment calldata rangePayment = rangePayments[i];
+            bytes32 rangePaymentHash = _payForRange(rangePayment);
+            emit RangePaymentCreated(msg.sender, rangePaymentHash, rangePayment);
+        }
+    }
+
+    /**
+     * @notice similar to `payForRange` except the payment is split amongst *all* stakers
+     * rather than just those delegated to operators who are registered to a single avs
+     */
+    function payAllForRange(
+        RangePayment[] calldata rangePayments
+    ) external onlyWhenNotPaused(PAUSED_PAY_ALL_FOR_RANGE) onlyPayAllForRangeSubmitter {
+        for (uint256 i = 0; i < rangePayments.length; i++) {
+            RangePayment calldata rangePayment = rangePayments[i];
+            bytes32 rangePaymentHash = _payForRange(rangePayment);
+            emit RangePaymentForAllCreated(msg.sender, rangePaymentHash, rangePayment);
+        }
+    }
+
+    function _payForRange(RangePayment calldata rangePayment) internal returns (bytes32) {
+        require(rangePayment.strategiesAndMultlipliers.length > 0, "PaymentCoordinator.payForRange: no strategies set");
+        require(rangePayment.amount > 0, "PaymentCoordinator.payForRange: amount cannot be 0");
+        require(
+            rangePayment.duration <= MAX_PAYMENT_DURATION,
+            "PaymentCoordinator.payForRange: duration exceeds MAX_PAYMENT_DURATION"
+        );
+        require(
+            rangePayment.duration % calculationIntervalSeconds == 0,
+            "PaymentCoordinator.payForRange: duration must be a multiple of calculationIntervalSeconds"
+        );
+        // If retroactive payments enabled must at least be past LOWER_BOUND_START_RANGE,
+        // otherwise should start earliest at current block timestamp
+        require(
+            (retroactivePaymentsEnabled && rangePayment.startTimestamp >= LOWER_BOUND_START_RANGE) ||
+                rangePayment.startTimestamp >= block.timestamp,
+            "PaymentCoordinator.payForRange: invalid startTimestamp set"
+        );
+
+        // Require rangePayment is for whitelisted strategies
+        for (uint256 j = 0; j < rangePayment.strategiesAndMultlipliers.length; j++) {
+            IStrategy strategy = rangePayment.strategiesAndMultlipliers[j].strategy;
+            require(
+                strategyManager.strategyIsWhitelistedForDeposit(strategy),
+                "PaymentCoordinator.payForRange: strategy not whitelisted for deposit"
+            );
+        }
+
+        // Set hash of rangePayment in mapping
+        bytes32 rangePaymentHash = keccak256(abi.encode(msg.sender, rangePayment));
+        isRangePaymentHash[msg.sender][rangePaymentHash] = true;
+
+        rangePayment.token.safeTransferFrom(msg.sender, address(this), rangePayment.amount);
+        return rangePaymentHash;
+    }
+
+    /**
+     * @notice Claim payments for the given claim
+     * @param claim The claims to be processed
+     */
```

Comment:
Is this method more thoroughly documented in the interface?

Reply:
https://github.com/Layr-Labs/eigenlayer-contracts/pull/488/commits/59612913dd174b2cc932c15b5e9ff1b950f68f55
Added some additional documentation